lineage/CellVar.py

```diff
@@ -0,0 +1,106 @@
+        self.left = CellVar(state=left_state, left=None, right=None, parent=self, gen=self.gen + 1)  # assign the resulting states to new cells
+        self.right = CellVar(state=right_state, left=None, right=None, parent=self, gen=self.gen + 1)  # ensure that those cells are related
+
+        return self.left, self.right
+
+    def _isParent(self):
+        """ Boolean. Returns true if the cell has daughters. """
+        return self.left is not None or self.right is not None
+
+    def _isChild(self):
+        """ Boolean. Returns true if this cell has a known parent. """
+        if self.parent:
+            return self.parent._isParent()
+        else:
+            return False
+
+    def _isRootParent(self):
+        """ Boolean. Returns true if this cell is the first cell in a lineage. """
+        bool_parent = False
+        if not self.parent and self.gen == 1:
+            bool_parent = True
+        return bool_parent
+
+    def _isLeaf(self):
+        """ Boolean. Returns true when a cell is a leaf with no children. """
+        return self.left is None and self.right is None
+
+    def _get_sister(self):
+        """ Member function that gets the sister of the current cell. """
+        cell_to_return = None
+        if self.parent.left is self:
+            cell_to_return = self.parent.right
+        elif self.parent.right is self:
+            cell_to_return = self.parent.left
+        return cell_to_return
+
+    def _get_root_cell(self):
+        """ Get the first cell in the lineage to which this cell belongs. """
+        curr_cell = self
+        while curr_cell.gen > 1:
+            curr_cell = curr_cell.parent
+        assert curr_cell._isRootParent()
+        return curr_cell
+    
+    def _get_daughters(self):
+        """ Get the left and right daughters of a cell. """
+        temp = []
+        if self.left is not None:
+            temp.append(self.left)
+        if self.right is not None:
+            temp.append(self.right)
+        return temp
+
+    def __repr__(self):
+        if hasattr(self, 'obs'):
+            return "Generation: {}, State: {}, Observation: {}".format(self.gen, self.state, self.obs)
+        else:
+            return "Generation: {}, State: {}, Observation: {}".format(self.gen, self.state, "This cell has no observations to report.")
+
+    def __str__(self):
+        if hasattr(self, 'obs'):
+            return "Generation: {}, State: {}, Observation: {}".format(self.gen, self.state, self.obs)
+        else:
+            return "Generation: {}, State: {}, Observation: {}".format(self.gen, self.state, "This cell has no observations to report.")
+
+
+def _double(parent_state, T):
+    """ Function that essentially rolls two of the same loaded dice given a state that determines the row of the transition matrix. The results of the roll of the loaded dice are two new states that are returned. """
```

Comment:
what are the two same loaded die you are referring to and what are the _two new states that are returned_ ?

Reply:
i see youre initializing the T for this cell.

Reply:
actually, can u explain this for me

Reply:
Drawing samples from a multinomial distribution can be thought of as rolling a loaded die. It's the same way as drawing samples from a bernoulli distribution can be thought of as flipping a loaded coin. Each row of the T matrix represents a loaded die where the "loading" is the probability that the "die" lands on that state. We roll this loaded die (sample the multinomial distribution twice) when considering what the states are going to be for the two daughter cells from a parent cell of state_j. We go to the j_th row, use that row to load the die (to tell the multinomial distribution the probability of each state), roll the die (sample the distribution) and the two results we get are the states of the two daughters.

Reply:
lmk if this makes sense or not, or should we add more documentation

Reply:
ah, use that row to roll the die is based on what state you are in determines the probabilities of transitioning ie of the die showing a particular face

Reply:
`are the two new states` **of the daughter cells** that are returned is what you mean right.